client/src/lib/components/NewTransactionDialog.svelte

```diff
@@ -0,0 +1,138 @@
+<script lang="ts">
+	import Textfield, { HelperLine } from '@smui/textfield';
+	import Button, { Label } from '@smui/button';
+	import { form, field } from 'svelte-forms';
+	import { required } from 'svelte-forms/validators';
+	import Dialog, { Content, Title } from '@smui/dialog';
+	import { CurrencyType, NewTransaction } from '$mock/api/models/expense_model';
+	import { createEventDispatcher } from 'svelte';
+	import Select, { Option } from '@smui/select';
+
+	export let expenseId: number;
+	export let currencyTypes: CurrencyType[];
+
+	let open = false;
+
+	const dispatch = createEventDispatcher<{ newTransaction: { transaction: NewTransaction } }>();
+
+	const donorName = field('donorName', '', [required()]);
+	const selectedCurrencyTypeId = field('selectedCurrencyTypeId', -1, [required()]);
+	const value = field<number | undefined>('value', 0, [required()]); // select component makes this undefined when nothing is selected
+	const date = field('date', '', [required()]);
+	const newTransactionForm = form(donorName, value, date);
+
+	function onSubmit() {
+		if (
+			$newTransactionForm.valid &&
+			$donorName.value &&
+			selectedCurrencyTypeId &&
+			$value.value &&
+			$date.value
```

Comment:
Is it possible for the form to be valid but these values are missing?
I would have probably only check the first one.

Reply:
For sure, I was a little blind for the inner workings of the validator library, it is much more streamlined now.

Reply:
Yes I was a little blind about the inner workings of the validator library, it is more clean now.